src/policy/mod.rs

```diff
@@ -0,0 +1,274 @@
+//! Zone policy.
+
+use std::{fs, io, sync::Arc, time::Duration};
+
+use bytes::Bytes;
+use camino::Utf8PathBuf;
+use domain::base::Name;
+
+use crate::config::Config;
+
+pub mod file;
+
+//----------- Policy -----------------------------------------------------------
+
+/// A policy.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Policy {
+    /// The latest version of the policy.
+    pub latest: Arc<PolicyVersion>,
+
+    /// Whether the policy is being deleted.
+    ///
+    /// This is an intermediate state used to prevent race conditions while the
+    /// policy is being removed.  In this state, new zones cannot be attached to
+    /// this policy.
+    pub mid_deletion: bool,
+
+    /// The zones using this policy.
+    pub zones: foldhash::HashSet<Name<Bytes>>,
```

Comment:
Is zone name enough? I'm wondering if we need to know that an already signed zone used this policy when we (incrementally) resign it, while if the policy is changed for the same zone would the next unsigned version of the zone used the new policy while the previously signed version refers to the old policy?